infra/src/main/java/com/redhat/service/smartevents/infra/models/processors/ProcessorDefinition.java

```diff
@@ -18,6 +19,9 @@ public class ProcessorDefinition {
     @JsonProperty("requestedAction")
     private Action requestedAction;
 
+    @JsonProperty("requestedSource")
```

Comment:
I'll have a proper look next week but biggest gripe is that this change really suggests we need a base `ProcessorDefinition` abstract class with two sub-classes; one for Sinks (with the Action field) and the other for Sources (with the Source field). That'd obviously have impact elsewhere.

Reply:
In theory this makes sense, but practically it would require quite some changes in the code that are not worth at this point. We can definitely think about this if the `ProcessorDefinition` becomes more complex.

Reply:
@kostola from @manstis review message, I would say, we need a jira to track this

Reply:
@manstis No I see your point, and happy to take a follow up issue into the backlog, but at this point I'm inclined to agree with @kostola in terms of effort vs reward of doing the work. 

In particular if we have dedicated constructors for the two ways in which a `ProcessorDefinition` can be created e.g one that requires `Source` and resolved `Action` and the other than requires requested `Action` and resolved `Action` then I think that helps to ensure we don't create the objects incorrectly.

cheers,

Rob

Reply:
@robpblake @manstis I created a follow-up task for this: [MGDOBR-635](https://issues.redhat.com/browse/MGDOBR-635)